legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-bigquery/src/main/java/org/finos/legend/engine/persistence/components/relational/bigquery/sql/BigQueryDataTypeMapping.java

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.persistence.components.relational.bigquery.sql;
+
+import org.finos.legend.engine.persistence.components.logicalplan.datasets.FieldType;
+import org.finos.legend.engine.persistence.components.relational.bigquery.sqldom.schema.Bool;
+import org.finos.legend.engine.persistence.components.relational.bigquery.sqldom.schema.Bytes;
+import org.finos.legend.engine.persistence.components.relational.bigquery.sqldom.schema.Float64;
+import org.finos.legend.engine.persistence.components.relational.bigquery.sqldom.schema.Json;
+import org.finos.legend.engine.persistence.components.relational.bigquery.sqldom.schema.String;
+import org.finos.legend.engine.persistence.components.relational.sql.DataTypeMapping;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.DataType;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.Date;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.DateTime;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.Int64;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.Numeric;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.Time;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.Timestamp;
+import org.finos.legend.engine.persistence.components.relational.sqldom.schema.VariableSizeDataType;
+
+
+public class BigQueryDataTypeMapping implements DataTypeMapping
+{
+    public DataType getDataType(FieldType type)
+    {
+        VariableSizeDataType dataType;
+        switch (type.dataType())
+        {
+            // Numeric Data Types
+            case INT:
+            case INTEGER:
+            case BIGINT:
+            case TINYINT:
+            case SMALLINT:
+            case INT64:
+                dataType = new Int64();
```

Comment:
What is the rational to fit everything on 64 bit? Seems as something that will lead to incorrect data.

Im not sure how these mappings are used, but if we allow 64 bit on some places that then we try to insert into the database into TinyInt, we might end with bit overflows, and we might not know the value on the DB will be wrong.

Thoughts?

Reply:
This mapping is for mapping logical data types to big query data types.
In Big Query, INT, SMALLINT, INTEGER, BIGINT, TINYINT, and BYTEINT are aliases for INT64.
So we have mapped all of them to INT64.
https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#numeric_types